test/new-e2e/test-infra-definition/cross_platform_test.go

```diff
@@ -34,36 +36,27 @@ func (s *crossPlatformSuite) TestWindows() {
 	// As the Windows installer cannot downgrade an agent version,
 	// the order of tests matter. New tests must be added at
 	// the end of the function.
-	os := ec2os.WindowsOS
+	os := os.WindowsDefault
 	s.subTestInstallAgentVersion(os)
 	s.subTestInstallAgent(os)
 }
 
 func (s *crossPlatformSuite) TestOtherOSES() {
-	for _, os := range ec2os.GetOSTypes() {
-		if os == ec2os.WindowsOS || os == ec2os.UbuntuOS {
-			// Windows and Ubuntu have their dedicated tests
-			continue
-		}
-
-		if os == ec2os.CentOS || os == ec2os.RockyLinux {
-			// TODO: There is a bug for CentOS and RockyLinux
-			continue
-		}
-
+	// TODO: There is a bug for CentOS and RockyLinux
+	for _, os := range []os.Descriptor{os.AmazonLinuxDefault, os.DebianDefault, os.FedoraDefault, os.RedHatDefault, os.SuseDefault} {
 		s.subTestInstallAgent(os)
 	}
 }
 
-func (s *crossPlatformSuite) subTestInstallAgent(os ec2os.Type) {
+func (s *crossPlatformSuite) subTestInstallAgent(os os.Descriptor) {
 	s.T().Run(fmt.Sprintf("Test install agent %v", os), func(t *testing.T) {
-		s.UpdateEnv(e2e.AgentStackDef(e2e.WithVMParams(ec2params.WithOS(os))))
-		output := s.Env().Agent.Status().Content
+		s.UpdateEnv(awsvm.Provisioner(awsvm.WithEC2VMOptions(ec2.WithOS(os))))
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
We should change this test to follow guidelines of cross-platform tests